client/manager_test_helpers.go

```diff
@@ -121,27 +153,48 @@ func TestHelperCreateGetUpdateDeleteClient(k string, m Storage) func(t *testing.
 			BackChannelLogoutSessionRequired:  true,
 		}
 
-		require.NoError(t, m.CreateClient(ctx, c))
-		assert.Equal(t, c.GetID(), "1234")
+		require.NoError(t, t1.CreateClient(ctx, t1c1))
+		{
+			t2c1 := *t1c1
+			require.Error(t, t2.CreateClient(ctx, &t2c1), "should not be able to create the same client in other manager/tenant; are they backed by the same database?")
```

Comment:
Good catch! This code shouldn't work because we exclude `pk` in the Create function. The reason why it does work is that `pk` is a UUID-typed primary key, leading to [this](https://github.com/gobuffalo/pop/blob/c285511d17469aa430ab08202a6f2353c3ae19e5/dialect_common.go#L81) line that adds `pk` despite its exclusion from `cols`.

This is the only place in Hydra where we rely on `Create` to exclude a field in order to generate a random ID, so I added `c.ID = uuid.Must(uuid.NewV4())` in CreateClient persister + a test.